src/main/java/com/boomi/flow/services/boomi/mdh/match/MatchResponseMapper.java

```diff
@@ -0,0 +1,99 @@
+
+        var propertiesMatched = new Property(FuzzyMatchDetialsConstants.MATCH, new ArrayList<>());
+        var propertiesDuplicated = new Property(FuzzyMatchDetialsConstants.DUPLICATE, new ArrayList<>());
+        var propertiesAlreadyLinked = new Property(FuzzyMatchDetialsConstants.ALREADY_LINKED, new ArrayList<>());
+        var object = new MObject(universeId + " match", externalId, Arrays.asList(propertiesMatched, propertiesDuplicated, propertiesAlreadyLinked));
+        object.setTypeElementBindingDeveloperName(object.getDeveloperName());
+
+        var properties =  new ArrayList<Property>();
+        result.getEntity().get(universe.getName())
+                .forEach((key, value) -> properties.add(new Property(key, (String) value)));
+
+        properties.add(new Property(GoldenRecordConstants.SOURCE_ID_FIELD, result.getIdResource()));
+        properties.add(new Property(FuzzyMatchDetialsConstants.FUZZY_MATCH_DETAILS, (MObject) null));
+        properties.addAll(object.getProperties());
+
+        object.setProperties(properties);
+
+        if ("SUCCESS".equals(result.getStatus())) {
+            result.getMatch().forEach(match -> addMatchesToProperty(propertiesMatched, universe, match, universe.getIdField(), true));
+            result.getDuplicate().forEach(match -> addMatchesToProperty(propertiesDuplicated, universe, match, universe.getIdField(), true));
+        } else if ("ALREADY_LINKED".equals(result.getStatus())) {
+            var entityH = result.getEntity().get(universe.getName());
+            addAlreadyLinked(propertiesAlreadyLinked, universe, entityH, universe.getIdField());
+        }
+
+        return object;
+    }
+
+
+    private static void addAlreadyLinked(Property alreadyLinkedProperty, Universe universe, Map<String, Object> entity, String idField){
+        var object = new MObject(universe.getId().toString() + " match");
+        object.setExternalId(entity.get(idField).toString());
+
+        entity.forEach((key, value) -> {
+            var property = new Property(key, value);
+            object.getProperties().add(property);
+        });
+
+        alreadyLinkedProperty.getObjectData().add(object);
+    }
+
+    private static void addMatchesToProperty(Property propertyMatches, Universe universe, Map<String, Object> matchResults, String idField, boolean addFuzzyMatchDetails){
+        var entity = (Map<String, Object>) matchResults.get(universe.getName());
+        var object = new MObject(universe.getId().toString() + " match");
+        object.setTypeElementBindingDeveloperName(object.getDeveloperName());
+
+        object.setExternalId(entity.get(idField).toString());
+
+        entity.forEach((key, value) -> {
+            var property = new Property(key, value);
+            object.getProperties().add(property);
+        });
+
+        if (addFuzzyMatchDetails) {
+            addFuzzyMatchDetails(object, (HashMap<String, Object>) matchResults.get("fuzzyMatchDetails"));
```

Comment:
This cast should probably just be to `Map<String, Object>` rather than assuming it's only one of potentially multiple `Map` implementations